diff --git a/modules/CiteProc/biblio_citeproc.admin.inc b/modules/CiteProc/biblio_citeproc.admin.inc
index 2ee9afd..cf16a8e 100644
--- a/modules/CiteProc/biblio_citeproc.admin.inc
+++ b/modules/CiteProc/biblio_citeproc.admin.inc
@@ -7,21 +7,26 @@ function biblio_citeproc_style_manager_form($form, &$form_state) {
     $cache = cache_get('github_csl_repo');
     if (!$cache || $cache->expire < time()) {
       module_load_include('php', 'biblio_citeproc', '/Github/Autoloader');
-      Github_Autoloader::register();
-      $github = new Github_Client();
-      $branches = $github->getRepoApi()->getRepoBranches('citation-style-language', 'styles');
-      $tree = $github->getObjectApi()->listBlobs('citation-style-language', 'styles', $branches['master']);
-
-      $tree = array_flip($tree);
-      $options = array();
-      foreach ($tree as $sha => $file) {
-        if (strstr($file, '.csl')) {
-          $options[$sha] =  basename($file);
+      try {
+        Github_Autoloader::register();
+        $github = new Github_Client();
+        $branches = $github->getRepoApi()->getRepoBranches('citation-style-language', 'styles');
+        $tree = $github->getObjectApi()->listBlobs('citation-style-language', 'styles', $branches['master']);
+
+        $tree = array_flip($tree);
+        $options = array();
+        foreach ($tree as $sha => $file) {
+          if (strstr($file, '.csl')) {
+            $options[$sha] =  basename($file);
+          }
+        }
+        if (!empty($options)) {
+          $expire = time()+86400; //one day from now
+          cache_set('github_csl_repo', $options, 'cache', $expire);
         }
       }
-      if (!empty($options)) {
-        $expire = time()+86400; //one day from now
-        cache_set('github_csl_repo', $options, 'cache', $expire);
+      catch (Exception $e) {
+        drupal_set_message('Error fetching list of styles.', 'warning');
       }
     }
     else {
